cpp/src/binaryop/compiled_binop2.cu

```diff
@@ -0,0 +1,147 @@
+#include <cudf/utilities/traits.hpp>
+#include <cudf/detail/utilities/device_operators.cuh>
+#include <cudf/detail/copy.hpp>
+
+#include <cudf/detail/iterator.cuh>
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+  //----
+namespace {
+  template<typename lhsT, typename rhsT, typename Op>
+  struct out_dispatcher {
+    mutable_column_device_view out;
+    column_device_view lhs;
+    column_device_view rhs;
+    Op op;
+    __host__ __device__ out_dispatcher(
+                mutable_column_device_view& out,
+                column_device_view const& lhs,
+                column_device_view const& rhs,
+                Op const op) : out(out), lhs(lhs), rhs(rhs), op(op) { }
+
+    template <typename outT>
+    static constexpr bool is_supported() { return
+        is_fixed_width<outT>() && (!std::is_same<outT, cudf::experimental::bool8>::value) && 
+        is_fixed_width<lhsT>() && (!std::is_same<lhsT, cudf::experimental::bool8>::value) && 
+        is_fixed_width<rhsT>() && (!std::is_same<rhsT, cudf::experimental::bool8>::value) && 
+        std::is_convertible<lhsT, outT>::value &&
+        std::is_convertible<rhsT, outT>::value;
+    }
+
+    template<typename outT,
+      std::enable_if_t<is_supported<outT>()>* = nullptr>
+    __device__ void operator()(size_type i)
+    {
+      (out.data<outT>())[i] = 
+               op(static_cast<outT>(lhs.element<lhsT>(i)),
+                  static_cast<outT>(rhs.element<rhsT>(i)));
+    }
+    template<typename outT,
+      std::enable_if_t<!is_supported<outT>()>* = nullptr>
+    __device__ void operator()(size_type i)
+    { }
+  };
+
+  template<typename rhsT, typename Op>
+  struct lhs_dispatcher {
+    mutable_column_device_view out;
+    column_device_view lhs;
+    column_device_view rhs;
+    Op op;
+    __host__ __device__ lhs_dispatcher(
+                mutable_column_device_view& out,
+                column_device_view const& lhs,
+                column_device_view const& rhs,
+                Op const op) : out(out), lhs(lhs), rhs(rhs), op(op) { }
+    template<typename lhsT>
+    __device__ void operator()(size_type i)
+    {
+      cudf::experimental::type_dispatcher(out.type(), out_dispatcher<lhsT, rhsT, Op>{out, lhs, rhs, op}, i);
+    }
+  };
+
+  template<typename Op>
+  struct rhs_dispatcher {
+    mutable_column_device_view out;
+    column_device_view lhs;
+    column_device_view rhs;
+    Op op;
+    __host__ __device__ rhs_dispatcher(
+                mutable_column_device_view const& out,
+                column_device_view const& lhs,
+                column_device_view const& rhs,
+                Op const op) : out(out), lhs(lhs), rhs(rhs), op(op) { }
+    template<typename rhsT>
+    __device__ void operator()(size_type i)
+    {
+      cudf::experimental::type_dispatcher(out.type(), lhs_dispatcher<rhsT, Op>{out, lhs, rhs, op}, i);
```

Comment:
I thought this was a gotcha and maybe fixing it would increase compile time. I thought the compiler would've made some optimization realizing two dispatches could be combined.
But surprisingly it didn't. Compile time is the same.

```suggestion
      cudf::experimental::type_dispatcher(lhs.type(), lhs_dispatcher<rhsT, Op>{out, lhs, rhs, op}, i);
```